### PR TITLE
KafkaSink ContentMode not checking

### LINKED ISSUE #2951: KafkaSinks ContentMode ignored
**Describe the bug**

The default content mode when installing a Kafka Sink is `binary`:

https://github.com/knative-sandbox/eventing-kafka-broker/blob/9459144e95bd13df164340970a96b58a2761daa4/control-plane/pkg/apis/eventing/v1alpha1/kafka_sink_defaults.go#L28-L32

In #2932 we migrated an e2e test for the Sink, with a job for publishing to the Sink using `structured` content mode, without changing the default in Sink before. This test still succeeds, so 

* No verification is happening, or
* There is a bug in the sink implementation for changing content mode

 **To Reproduce**
Steps to reproduce the behavior.

Run the `TestBrokerTirggersSink` test from #2932

 **Knative release version**
On main branch (today with 5475bd8641cb0b732dca4ef1adc2c67d0d68e358)

### PR DESCRIPTION
**Pull request by debasishbsws**
Fixes #2951 

## Proposed Changes

- set the sink content mode to `structured`
- change the `Assert` to `IsDone()`, so that, the test fails if the job is uncompleted.

### CODE REVIEW

**Comment by debasishbsws** [comment]
Changed the [consumer.go](https://github.com/knative-sandbox/eventing-kafka-broker/blob/main/test/test_images/kafka-consumer/consumer.go) file. Now the pod get into the complete state after finishes executing without an error.

File: test/test_images/kafka-consumer/consumer.go [hunk 1]
```diff
@@ -110,10 +117,28 @@ func main() {
 				set.Delete(e.ID())
 				log.Printf("Event matching received: %s\nRemaining: %v\n", e.ID(), set)
 			} else {
-				log.Fatalf("Failed to match event: %s\n%v\n", e.String(), set.List())
+				log.Printf("Failed to match event: %s\n%v\n", e.String(), set.List())
```
[the remark this answers is not in the document]
Comment (debasishbsws) [reply test/test_images/kafka-consumer/consumer.go:120]: Actually, we are not only receiving 2 events, we are receiving multiple events with the same trigger ID like  
image

After receiving an event from a trigger we are deleting that id from the `set`. and if we subsequently receive an event from the same trigger before the other one a `fatal` was passed even though we may get the other trigger event later, which actually happens most of the time.

File: test/test_images/kafka-consumer/consumer.go [hunk 2]
```diff
@@ -110,10 +117,28 @@ func main() {
...
+		case <-isConsumerClose:
+			break L
+		}
+	}
+
+	if set.Len() > 0 {
+		log.Fatalf("Not all event consumed, Remaining ID: %v\n", set.List())
+	} else {
+		log.Println("All events consumed")
+	}
```
[the remark this answers is not in the document]
Comment (debasishbsws) [reply test/test_images/kafka-consumer/consumer.go:141]: Yes, That's right I have put that for debug logging purposes but the condition is not required now. let me remove that

File: test/test_images/kafka-consumer/consumer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pierDipi) [reply test/test_images/kafka-consumer/consumer.go:120]: I see, that makes sense to me

**Comment by pierDipi** [comment]
I see some related tests are failing on https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka-broker/2998/integration-tests_eventing-kafka-broker_main/1633328537492525056

**Comment by debasishbsws** [comment]
@pierDipi to run the test should I run this? 
```
SYSTEM_NAMESPACE=knative-eventing BROKER_CLASS=Kafka go test -tags=e2e  -run ^TestKafkaSinkV1Alpha1DefaultContentMode$ knative.dev/eventing-kafka-broker/test/e2e_sink
```

**Comment by debasishbsws** [comment]
Should I include the files generated by `hack/update-codegen.sh` into pr or not?

**Comment by debasishbsws** [comment]
A clean and clear PR #3016 is been made a substitution for this one. For some mistake, this branch has gotten some cluttered commits and tests are failing.

**Pull request closed without merging**